### PR TITLE
SDV2-1061  PHP 7.1

### PR DESCRIPTION
**Pull request by webit4me**
Add an image containing PHP 7.1, installed using `ppa:ondrej/php` personal package archives

### CODE REVIEW

File: opg-php-fpm-71-ppa-1604/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                  php7.1-curl \
+                  php7.1-fpm \
+                  php7.1-imagick \
+                  php7.1-mbstring \
+                  php7.1-mcrypt \
+                  php7.1-mongo \
+                  php7.1-pgsql \
+                  php7.1-xml"
+
+RUN apt-get install -y python-software-properties && \
```
Comment (brettminnie) [inline opg-php-fpm-71-ppa-1604/Dockerfile]: you would want to do an apt-update before this install to ensure you are getting the latest package

File: opg-php-fpm-71-ppa-1604/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    add-apt-repository -y ppa:ondrej/php && \
+    apt-get update -y && \
+    apt-get install $PHP_PACKAGES -y && \
+    apt clean && apt -y autoremove && \
+    rm -rf /var/lib/cache/* /var/lib/log/* /tmp/* /var/tmp/* && \
+    cd /tmp && \
+    curl -sS https://getcomposer.org/installer | php && \
+    mv composer.phar /usr/local/bin/composer
+
+RUN rm -f /etc/php/7.1/fpm/php-fpm.conf /etc/php/7.1fpm/pool.d/www.conf /etc/cron.d/php && \
```
Comment (brettminnie) [inline opg-php-fpm-71-ppa-1604/Dockerfile]: please combine this `RUN` with the one above so we have one layer

File: opg-php-fpm-71-ppa-1604/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    mkdir -p /data/session
+
+ADD  docker/confd /etc/confd
+
+ADD  docker/service/php-fpm /etc/sv/php-fpm
+RUN  chmod a+x /etc/sv/php-fpm/run && \
+     ln -s /etc/sv/php-fpm /etc/service/
+
+RUN  echo "<?php echo phpinfo(); ?>" > /app/public/info.php && \
```
Comment (brettminnie) [inline opg-php-fpm-71-ppa-1604/Dockerfile]: combine these `RUN` commands

File: opg-php-fpm-71-ppa-1604/docker/my_init.d/40-opg_php_xdebug_remote_host [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+
+rm -f /etc/php/7.1/fpm/conf.d/*xdebug*
```
Comment (brettminnie) [inline opg-php-fpm-71-ppa-1604/docker/my_init.d/40-opg_php_xdebug_remote_host]: is this the correct path for the php fpm install? normally it lives under php/major_version_number/fpm

File: opg-php-fpm-71-ppa-1604/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,75 @@
+FROM registry.service.opg.digital/opg-nginx-1604
+
+ENV PHP_PACKAGES="mcrypt \
```
Comment (brettminnie) [inline opg-php-fpm-71-ppa-1604/Dockerfile]: please change this from ENV to ARG, we don't need to expose these to the environment

File: opg-base-1604/docker/my_init.d/99-generate-ssl [hunk 6]
```diff
@@ -17,7 +17,8 @@ cp key.pem key.pem.orig
 openssl rsa -passin pass:x -in key.pem.orig -out key.pem
 
 # Generating certificate signing request
-openssl req -new -key key.pem -out cert.csr -subj "/C=GB/ST=GB/L=London/O=OPG/OU=Digital/CN=default"
+openssl req -new -key key.pem -out cert.csr -subj \
+  "/C=GB/ST=GB/L=London/O=OPG/OU=Digital/CN=${SSL_CERT_COMMON_NAME:=default}"
```
Comment (brettminnie) [inline opg-base-1604/docker/my_init.d/99-generate-ssl]: please rebase master as this change now exists there

File: opg-php-fpm-71-ppa-1604/Makefile [hunk 7]
```diff
@@ -0,0 +1,9 @@
+.PHONY: build
+
+registryUrl ?= registry.service.opg.digital
+
+image = opg-php71-fpm-1604
```
Comment (brettminnie) [inline opg-php-fpm-71-ppa-1604/Makefile]: this name needs to match the directory name for the master makefile to work correctly

File: opg-php-fpm-71-ppa-1604/docker/my_init.d/40-opg_php_xdebug_remote_host (hunk 4 above)
Comment (webit4me) [reply opg-php-fpm-71-ppa-1604/docker/my_init.d/40-opg_php_xdebug_remote_host]: yea, i can confirm, expected files are under the same path in containers build from this image 
```
root@d21ef9abbc89:/app# ls -la /etc/php/7.1/
total 28
drwxr-xr-x 11 root root 4096 Nov 17 13:41 .
drwxr-xr-x 10 root root 4096 Nov 17 13:41 ..
drwxr-xr-x  5 root root 4096 Nov 17 13:41 cli
drwxr-xr-x  7 root root 4096 Nov 17 13:41 fpm
drwxr-xr-x  2 root root 4096 Nov 17 13:30 mods-available
-rw-r--r--  1 root root   98 Nov 16 14:10 xdebug-disable.ini
-rw-r--r--  1 root root  334 Nov 16 14:10 xdebug-enable.ini
```

File: opg-php-fpm-71-ppa-1604/docker/confd/templates/app.conf.tmpl [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    {{ else }}
+    set $index "index.php";
+    {{ end }}
+
+    index $index;
+
+    #We've found some php client libraries not understanding chunked response
+    chunked_transfer_encoding off;
+
+ #    fastcgi_buffers 256 4k;
```
Comment (carinadigital) [inline opg-php-fpm-71-ppa-1604/docker/confd/templates/app.conf.tmpl]: If we are commenting this, Why not remove it?

File: opg-php-fpm-71-ppa-1604/docker/my_init.d/40-update-php-fpm-environment [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+"""
+import os
+
+skip_variables = ["LS_COLORS"]
+
+
+with open("/etc/php/7.1/fpm/pool.d/app.conf", "a") as app_conf:
+
+    for name, value in os.environ.iteritems():
```
Comment (carinadigital) [inline opg-php-fpm-71-ppa-1604/docker/my_init.d/40-update-php-fpm-environment]: Are we sure we want all env values passed through to PHP?

Comment (brettminnie) [reply opg-php-fpm-71-ppa-1604/docker/my_init.d/40-update-php-fpm-environment]: Think this has been a hangover from the original php-fpm scripts, we probably need to look at it in all php-fpm images

Comment (carinadigital) [reply opg-php-fpm-71-ppa-1604/docker/my_init.d/40-update-php-fpm-environment]: I think it's best to least this for now then. Extra env vars shouldn't be a problem.

Comment (brettminnie) [reply opg-php-fpm-71-ppa-1604/docker/my_init.d/40-update-php-fpm-environment]: Created a story in my current sprint to investigate it and possibly update

File: opg-php-fpm-71-ppa-1604/docker/confd/templates/app.conf.tmpl (hunk 8 above)
Comment (webit4me) [reply opg-php-fpm-71-ppa-1604/docker/confd/templates/app.conf.tmpl]: this one also is coming from the php-fpm image, just checked on vitor's (original container) and it seems to be commented out on our current setups as well. will remove them shortly

Comment (webit4me) [reply opg-php-fpm-71-ppa-1604/docker/confd/templates/app.conf.tmpl]: Done!

File: opg-php-fpm-71-ppa-1604/docker/my_init.d/40-update-php-fpm-environment (hunk 9 above)
Comment (webit4me) [reply opg-php-fpm-71-ppa-1604/docker/my_init.d/40-update-php-fpm-environment]: ah, I've just removed it as i couldnt see any relevant variable to services we are going to switch to use this container. if i notice any issue around missing env vars i will put the script back but will make the skip-list more explicit

**Pull request merged**